### PR TITLE
feat(listsADM): add canDeleteList route

### PR DESCRIPTION
**Pull request by mpro7**
resolves DEV-145

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/routing/admin/lists/DeleteListItemsRouteADM.scala [hunk 1]
```diff
@@ -63,4 +63,32 @@ class DeleteListItemsRouteADM(routeData: KnoraRouteData)
         )
     }
   }
+
+  /**
+   * Checks if a list (and its children) is unused and if can be deleted.
```
Comment (irinaschubert) [inline webapi/src/main/scala/org/knora/webapi/routing/admin/lists/DeleteListItemsRouteADM.scala:68]:
```suggestion
   * Checks if a list (and its children) can be deleted. This is the case if none of its nodes is used in data.
```

File: webapi/src/main/scala/org/knora/webapi/routing/admin/lists/DeleteListItemsRouteADM.scala [hunk 2]
```diff
@@ -63,4 +63,32 @@ class DeleteListItemsRouteADM(routeData: KnoraRouteData)
...
+   */
+  private def canDeleteList(featureFactoryConfig: FeatureFactoryConfig): Route =
+    path(ListsBasePath / "candelete" / Segment) { iri =>
+      get { requestContext =>
+        val listIri =
+          stringFormatter.validateAndEscapeIri(iri, throw BadRequestException(s"Invalid param list IRI: $iri"))
```
Comment (irinaschubert) [inline webapi/src/main/scala/org/knora/webapi/routing/admin/lists/DeleteListItemsRouteADM.scala:74]:
```suggestion
          stringFormatter.validateAndEscapeIri(iri, throw BadRequestException(s"Invalid list IRI: $iri"))
```

File: webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/canDeleteList.scala.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+@*
+ * Copyright © 2021 Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ *@
+
+@import org.knora.webapi.IRI
+
+@**
+ * Checks if a list (and its children) is unused and if can be deleted.
```
Comment (irinaschubert) [inline webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/canDeleteList.scala.txt:9]:
```suggestion
 * Checks if a list (and its children) is used in data.
```

File: webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/canDeleteList.scala.txt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ *@
+@(triplestore: String,
+  listIri: IRI)
+
+PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+
+SELECT DISTINCT ?isUsed
+
+WHERE {
+  BIND(IRI("@listIri") AS ?nodeToBeDeleted)
```
Comment (irinaschubert) [inline webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/canDeleteList.scala.txt:22]:
```suggestion
  BIND(IRI("@listIri") AS ?nodeToBeChecked)
```

File: webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/canDeleteList.scala.txt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  BIND(true AS ?isUsed)
+
+  {
+    ?nodeToBeDeleted knora-base:hasSubListNode* ?subNode .
```
Comment (irinaschubert) [inline webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/canDeleteList.scala.txt:26]:
```suggestion
    ?nodeToBeChecked knora-base:hasSubListNode* ?subNode .
```

**Review by irinaschubert — APPROVED** [review]
LGTM, just some suggestions regarding naming.

**Review by BalduinLandolt — APPROVED** [review]
looks good... I aggree with Irina's naming suggestions, and there is one typo (`unused` instead of `used`), but apart from that all fine

File: webapi/src/test/scala/org/knora/webapi/responders/admin/ListsResponderADMSpec.scala [hunk 6]
```diff
@@ -944,5 +944,79 @@ class ListsResponderADMSpec extends CoreSpec(ListsResponderADMSpec.config) with
...
+          iri = nodeInUseIri,
+          featureFactoryConfig = defaultFeatureFactoryConfig,
+          requestingUser = SharedTestDataADM.anythingAdminUser
+        )
+        val response: CanDeleteListResponseADM = expectMsgType[CanDeleteListResponseADM](timeout)
+        response.listIri should be(nodeInUseIri)
+        response.canDeleteList should be(false)
+      }
+
+      "return FALSE for a node that is unused but has a child which is used" in {
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/responders/admin/ListsResponderADMSpec.scala:961]: `node` -> `listIri`

**Review by subotic — CHANGES_REQUESTED** [review]

**Review by subotic — APPROVED** [review]

**Comment by subotic** [comment]
@mpro7 thanks, looks good now.

**Pull request merged**